picker/rofimoji.py

```diff
@@ -234,7 +235,7 @@ def open_main_rofi_window(self) -> Tuple[int, str]:
 
     def process_chosen_characters(self, chosen_characters: List[str]) -> str:
         processed_characters = ''.join(
-            self.add_skin_tone(line.split(' ')[0])
+            self.add_skin_tone(re.match(r'^[\u200e\u200f]?(?P<char>.)\s', line).group('char'))
```

Comment:
The regex is not correct yet, unfortunately, as some emojis can be composed of multiple characters (for example any "family" or "transgender flag").

After some playing, I think this works: `^[\u200e]?(?P<char>.?[^ ]*) .*`. Now multi-character emojis work again, and also whitespace (even ` ` itself!).
Can you please test it, too?

Reply:
Thank you for spotting this! Your regex fixes multi-character issues, but now the left-to-right mark inserts ` Left-To-Right` (without the mark). I'll keep playing with it and figure out one that encompasses all of these edge cases.

Reply:
(the `\s` is to show that it is preceded by a space. it does not insert a literal "\s".)

Reply:
Here's what I am using to determine what pattern to match. Feel free to update it if you can improve it. https://regex101.com/r/rgz2YT/2

Reply:
`^\u200e??(?P<char>.?[^ ]*) .*` seems to do the trick 🙂

Reply:
Unfortunately that now breaks the original issue we were trying to solve.
![image](https://user-images.githubusercontent.com/1340784/103873357-4a30e680-5095-11eb-8b12-299547637404.png)

Reply:
This is tricky... can you find something where `^(?:\u200e(?! ))?(?P<char>.[^ ]*) .*` doesn't work? 😉

But I'm wondering if it wouldn't be an easier solution to escape the ltr-mark in the source file by doubling it...

Reply:
Hmm... it seems to work just fine on [regex101](https://regex101.com/r/rgz2YT/3) but when I try it in code, Space returns nothing. I don't know why that is.

Reply:
I could imagine this failing if there's a ltr-mark set for space. Is it?

Reply:
There is not one.

Reply:
Hmm, I'm not sure why it wasn't before but I just tested it again and your solution works. I was using Sublime Text because I often forget to use `alt+p` in Firefox, but it's very much possible that Sublime has the same issue. In any case, I'll push the current solution for you can take a look at.